pkg/kubelet/kubelet.go

```diff
@@ -1310,10 +1310,12 @@ func (kl *Kubelet) cleanupOrphanedVolumes(pods []*api.Pod, runningPods []*kubeco
 			// TODO(yifan): Refactor this hacky string manipulation.
 			kl.volumeManager.DeleteVolumes(types.UID(parts[0]))
 			//TODO (jonesdl) This should not block other kubelet synchronization procedures
-			err := vol.TearDown()
-			if err != nil {
-				glog.Errorf("Could not tear down volume %q: %v", name, err)
-			}
+			go func(volumeName string) {
```

Comment:
E.g., what if it comes through next time and it's still working on tearing this down and it tears it down again.